### PR TITLE
[#3242] "Create" navbar item workflow update

### PR DESCRIPTION
**Pull request by Maurier**
#3242 Updates the design and workflow of "Create" navbar item as described in https://github.com/hydroshare/hydroshare/issues/3242#issuecomment-493268015

For large screens:
-----------------
The button:
image

The menu:
image

**For mobile devices:**
-----------------
image

**Modal titles for each resource type**
-----------------------------------------
Example:
image

### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [ ] Automated Testing

- [ ] Sufficient User and Developer Documentation

- [ ] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. Log into the site
2. Use the navbar "Create" button to create resources of each of the resource types.
3. Verify that the "Create" button and modal window look and behave as expected.

### CODE REVIEW

**Comment by Maurier** [comment]
@sblack-usu To review frontend changes. @pkdash to review backend changes.

File: hs_core/management/commands/set_bag_dirty.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+"""
```
Comment (sblack-usu) [inline hs_core/management/commands/set_bag_dirty.py]: I'm confused why these changes are in this pull.

Comment (Maurier) [reply hs_core/management/commands/set_bag_dirty.py]: I don't know how that ended up there, but I'll undo it.

Comment (sblack-usu) [reply hs_core/management/commands/set_bag_dirty.py]: I wonder if it's because I branched off of RC-1.23... maybe submit this PR against that branch?

File: theme/static/js/custom.js [hunk 2]
```diff
@@ -163,6 +158,17 @@ $(document).ready(function () {
...
+        let title = $(this).attr("data-modal-title");
+        let inputTitle = $(this).attr("data-modal-input-title");
+
+        $('#submit-title-dialog .modal-title').text(title);
+        $('#submit-title-dialog .modal-input-title').text(inputTitle);
+
+        $('#submit-title-dialog').modal('show');
+    });
+
+    $("#btn-resource-create").on("click", function () {
```
Comment (sblack-usu) [inline theme/static/js/custom.js]: I'll submit this change in a bit, but this is currently not passing the title to createresource.

File: hs_core/management/commands/set_bag_dirty.py (hunk 1 above)
Comment (sblack-usu) [reply hs_core/management/commands/set_bag_dirty.py]: heh, it was me.  I added it on accident. Removing it in another commit.
